modules/nextflow/src/test/groovy/nextflow/executor/BashWrapperBuilderTest.groovy

```diff
@@ -775,13 +777,20 @@ class BashWrapperBuilderTest extends Specification {
 
         when:
         def CONDA = Paths.get('/some/conda/env/foo')
-        binding = newBashWrapperBuilder(condaEnv: CONDA).makeBinding()
+        def config = new CondaConfig(['useMicromamba': MICROMAMBA])
+        binding = newBashWrapperBuilder(condaEnv: CONDA, condaConfig: config).makeBinding()
         then:
-        binding.conda_activate == '''\
+        binding.conda_activate == RESULT
+        where:
+        MICROMAMBA | RESULT
+        false      | '''\
                 # conda environment
-                source $(conda info --json | awk '/conda_prefix/ { gsub(/"|,/, "", $2); print $2 }')/bin/activate /some/conda/env/foo
+                source $(conda info --json | awk \'/conda_prefix/ { gsub(/"|,/, "", $2); print $2 }\')/bin/activate /some/conda/env/foo
+                '''.stripIndent()
+        true       | '''\
+                # conda environment
+                eval "$(micromamba shell hook --shell bash)" && micromamba activate /some/conda/env/foo
```

Comment:
better to add a new test for micromamba and keep the one for conda as it is

Reply:
Modified in last commit